wallet-new/src/Cardano/Wallet/Kernel/Actions.hs

```diff
@@ -55,13 +56,11 @@ data WalletWorkerState b = WalletWorkerState
 
 makeLenses ''WalletWorkerState
 
--- A helper function for lifting a `WalletActionInterp` through a monad transformer.
-lifted :: (Monad m, MonadTrans t) => WalletActionInterp m b -> WalletActionInterp (t m) b
-lifted i = WalletActionInterp
-    { applyBlocks  = lift . applyBlocks i
-    , switchToFork = \n bs -> lift (switchToFork i n bs)
-    , emit         = lift . emit i
-    }
+instance MFunctor WalletActionInterp where
+  hoist nat i = WalletActionInterp
+    { applyBlocks = fmap nat (applyBlocks i)
+    , switchToFork = fmap (fmap nat) (switchToFork i)
+    , emit = fmap nat (emit i) }
```

Comment:
When one sees `hoist`, `fmap`, `>>=` and the like, one can readily understand how something composes. Imagine if instead we had `changeTheMonadInFoo`, `modifyWhatIsInsideFoo` and `thisFooAfterThatFoo`: We'd have to research the behavior of these functions each time we wanted to use them.

Reply:
Uhm, I suspect this is subject to debate depending to who you ask; yours is surely a respectable point of view, but at the same time `lifted` here (or `hoist lift`) I don't think they are meant to be use so frequently in practice to justify the complexity brought in by this one.

To really play the devil's advocate, `lift` and any variant like `lifted` might be more familiar to the average Haskeller than `hoist`, which is something introduced fairly "recently" (say 3-4 years) in the Haskell ecosystem whereas `lift` & co have been around since Babylonian times.

But you have a point on `fmap` and `>>=`, I just feel that here what buys us is not enough as one is likely to use these functions sparingly and inside the wallet worker module, likely.

Reply:
@adinapoli-iohk and I just had a conversation. Ultimately, `MFunctor`, just like `Functor`, bring in *simplicity* at the cost of paying the upfront cost of understanding the laws of `fmap` and `hoist` once. Unless a majority can give a compelling reason why not to replace the ad-hoc implementation of `hoist` with a proper `MFunctor` instance, this instance will stay in.

Reply:
I'm in favor of using `mmorph` explicitly. It's a good abstraction with a clear set of laws, and I don't anticipate it will need to be used too much.

Reply:
> First of all, I think this is not a lawful use of MFunctor, because WalletActionInterp m isn't a Monad. 

It doesn't need to be, it needs to be a functor in the category of monads. And `WalletActionInterp m` is one of those. The handwavy basic requirement is that the kind of the functor is `(* -> *) -> * -> *`, and that `m` only ever shows up in positive positions inside `f`. Here's an example in the wild of something that is *not* a functor in the category of monads, even though its kind matches the expectation of `MFunctor`: https://hackage.haskell.org/package/di-monad-1.2/docs/Di-Monad.html#v:hoistDiT